lib/Dialect/Seq/Transforms/LowerSeqToSV.cpp

```diff
@@ -366,6 +371,17 @@ FirRegLower::RegLowerInfo FirRegLower::lower(FirRegOp reg) {
   // Move Attributes
   svReg.reg->setDialectAttrs(reg->getDialectAttrs());
 
+  // For array registers, we annotate ram_style attributes if
```

Comment:
Isn't this trying to catch something we don't know is being triggered?  All registers of vectors get the mapping seems a bit much without knowing that the use of the register is in a ram style which would be misinferred.

Reply:
Yes because we cannot exactly know the BRAM inference algorithm of vivado. So annotating vector registers should be conservatively enough.